linkcheck/checker/ftpurl.py

```diff
@@ -116,7 +118,9 @@ def cwd (self):
         Change to URL parent directory. Return filename of last path
         component.
         """
-        path = self.urlparts[2].encode(self.filename_encoding, 'replace')
+        path = self.urlparts[2]
+        if isinstance(path, bytes):
+            path = path.decode(self.filename_encoding, 'replace')
```

Comment:
Surely this is not only for Python 3 because it changes data types on Python 2 as well?

Reply:
ping? you said LGTM here, but should this be fixed?

Reply:
My "lgtm" was short for "I don't understand, but the tests pass, so why not?".

(I admit I felt better about the tests before I checked the coverage numbers and saw only 69%.)